qubesmanager/qube_manager.py

```diff
@@ -634,37 +812,39 @@ def on_domain_changed(self, vm, event, **_kwargs):
             return
 
         try:
-            self.vms_in_table[vm.qid].update(event=event)
+            self.qubes_cache.get_vm(qid=vm.qid).update(event=event)
+            self.proxy.invalidate()
         except exc.QubesPropertyAccessError:
             return  # the VM was deleted before its status could be updated
 
     def load_manager_settings(self):
-        for col in self.columns_indices:
-            col_no = self.columns_indices[col]
-            if col == 'Name':
+        # Load view menu
+        for col_no in range(len(self.qubes_model.columns_indices)):
+            column = self.qubes_model.columns_indices[col_no]
+            action = self.menu_view.addAction(column)
+            action.setCheckable(True)
+            action.toggled.connect(partial(self.showhide_column, col_no))
+            if column == 'Name':
                 # 'Name' column should be always visible
-                self.columns_actions[col_no].setChecked(True)
+                action.setChecked(True)
             else:
-                visible = self.manager_settings.value(
-                    'columns/%s' % col,
+                visible = self.manager_settings.value('columns/%s' % column,
                     defaultValue="true")
-                self.columns_actions[col_no].setChecked(visible == "true")
+                action.setChecked(visible == "true")
+                self.showhide_column(col_no, visible == "true")
 
-        self.sort_by_column = str(
-            self.manager_settings.value("view/sort_column",
-                                        defaultValue=self.sort_by_column))
-        self.sort_order = QtCore.Qt.SortOrder(
-            self.manager_settings.value("view/sort_order",
-                                        defaultValue=self.sort_order))
+        self.menu_view.addSeparator()
+        self.menu_view.addAction(self.action_toolbar)
+        self.menu_view.addAction(self.action_menubar)
 
-        try:
-            self.table.sortItems(self.columns_indices[self.sort_by_column],
-                                 self.sort_order)
-        except KeyError:
-            # the manager was sorted on a column that does not exist in the
-            # current version; possible only with downgrades
-            self.table.sortItems(self.columns_indices["Name"],
-                                 self.sort_order)
+        # Restore sorting
+        sort_column = int(self.manager_settings.value("view/sort_column"))
```

Comment:
This (and some other places in this function) fails if setting wasn't there  (`int(None)`), use `defaultValue=` argument.